File: data/src/main/java/com/icehousecorp/maunorafiq/data/city/repository/datasource/DiskCityDataStore.java
package com.icehousecorp.maunorafiq.data.city.repository.datasource;

import android.util.Log;

import com.icehousecorp.maunorafiq.data.city.disk.RealmService;
import com.icehousecorp.maunorafiq.data.city.entity.CityEntity;

import java.util.List;

/**
 * Created by maunorafiq on 12/1/16.
 */

public class DiskCityDataStore implements CityDataStore {

    private final String TAG  = this.getClass().getSimpleName();

    private final RealmService realmService;

    public DiskCityDataStore(RealmService realmService) {
        this.realmService = realmService;
    }

    @Override
    public List<CityEntity> getListCityEntity() {
        return this.realmService.get();
    }

    @Override
    public boolean putCityEntity(String city) {
        return this.realmService.put(city);
    }
}

File: data/src/main/java/com/icehousecorp/maunorafiq/data/city/repository/datasource/DiskCityDataStore.kt
package com.icehousecorp.maunorafiq.data.city.repository.datasource

import android.util.Log

import com.icehousecorp.maunorafiq.data.city.disk.RealmService
import com.icehousecorp.maunorafiq.data.city.entity.CityEntity

/**
 * Created by maunorafiq on 12/1/16.
 */

class DiskCityDataStore(private val realmService: RealmService) : CityDataStore {

    private val TAG = this.javaClass.simpleName

    override val listCityEntity: List<CityEntity>
        get() = this.realmService.get()

    override fun putCityEntity(city: String): Boolean {
        return this.realmService.put(city)
    }
}
